tsl/src/remote/dist_ddl.c

```diff
@@ -368,19 +368,24 @@ dist_ddl_process_drop(const ProcessUtilityArgs *args)
 {
 	DropStmt *stmt = castNode(DropStmt, args->parsetree);
 
+	/* For DROP TABLE and DROP SCHEMA operations hypertable_list will be empty */
 	if (list_length(args->hypertable_list) == 0)
 	{
-		/*
-		 * CASCADE operations of DROP TABLE and DROP SCHEMA are handled in
-		 * sql_drop trigger.
-		 */
-
-		/* For DROP TABLE and DROP SCHEMA operations hypertable_list will be
-		 * empty. Wait for sql_drop events.
-		 */
-		if (stmt->removeType == OBJECT_TABLE || stmt->removeType == OBJECT_SCHEMA)
-			dist_ddl_state_schedule(DIST_DDL_EXEC_ON_END, args);
-
+		switch (stmt->removeType)
+		{
+			case OBJECT_TABLE:
+				/* Wait for further sql_drop events  */
+				dist_ddl_state_schedule(DIST_DDL_EXEC_ON_END, args);
+				break;
+			case OBJECT_SCHEMA:
+				/* Forward DROP SCHEMA command to all data nodes, following
+				 * sql_drop events will be ignored */
```

Comment:
A completely clueless question: why do we have to execute DROP SCHEMA immediately, in contrast to DROP TABLE?

Reply:
Previously we did not support `CREATE SCHEMA` explicitly on data nodes and when`DROP SCHEMA CASCADE` was executed, we only shipped it by analyzing distributed hypertables which we part of that schema. No it is much simpler, we assume schema exists on all data nodes and we can drop it right away.